library/core/src/ptr/non_null.rs

```diff
@@ -1301,6 +1301,8 @@ impl<T: ?Sized> NonNull<T> {
     #[must_use]
     #[stable(feature = "pointer_is_aligned", since = "1.79.0")]
     #[rustc_const_unstable(feature = "const_pointer_is_aligned", issue = "104203")]
+    #[requires(ub_checks::can_dereference(self.as_ptr()))]
```

Comment:
This is actually a safe operation and it should be OK even if the pointer cannot be read.

Reply:
i.e.: can you please remove this pre-condition. The pointer is not in fact dereferenced in this function, thus, this should be safe for any pointer, including dangling and null.

Reply:
removed